ClassPaulita/GuessTheCardGame.cs

```diff
@@ -1,107 +1,174 @@
+            {
+                return true;
+            }
 
-        string userChoice = Console.ReadLine();
+            return false;
+        }
 
-        while (!IsValidInput(userChoice))
+        private string PrintMenu()
         {
-            Console.WriteLine("\n\tOoooops! You pressed an invalid option. Please try again.");
+
+            string menu = $"{spacing}Press P To Play" +
+                          $"{spacing}Press I For Instructions" +
+                          $"{spacing}Press B To See The High Score Board" +
+                          $"{spacing}Press Q To Quit";
 
             Console.WriteLine(menu);
 
-            userChoice = Console.ReadLine();
-        }
+            string userChoice = Console.ReadLine();
 
-        return userChoice;
-    }
+            while (!IsUserChoiceValid(userChoice))
+            {
+                Console.WriteLine("\n\tOoooops! You pressed an invalid option. Please try again.");
 
-    public void Run()
-    {
-        while (true)
-        {
-            string userChoice = PrintMenu();
+                Console.WriteLine(menu);
+
+                userChoice = Console.ReadLine();
+            }
+
+            return userChoice;
+        }
 
-            if (UserChoseToPlay(userChoice[0]))
+        public void Run()
+        {
+            while (true)
             {
-                Play();
+                string userChoice = PrintMenu();
+
+                if (UserChoseToPlay(userChoice[0]))
+                {
+                    Play();
+                }
+
+                if (UserchoseToSeeInstructions(userChoice[0]))
+                {
+                    Instructions();
+                }
+
+                if (UserChosetoSeeBoard(userChoice[0]))
+                {
+                    HighScoreBoard();
+                }
+
+                if (UserChoseToQuit(userChoice[0]))
+                {
+                    return;
+                }
             }
+        }
+
+        private void Play()
+        {
+            Console.WriteLine($"{spacing}Play!");
+
+            DeckOfCards deck = new DeckOfCards();
+            deck.Suffle();
+
+            Card previousCard = deck.Deal();
 
-            if (UserchoseToSeeInstructions(userChoice[0]))
+            int userScore = 0;
+
+
+            while (deck.CardsLeftOnDeck > 0)
             {
-                Instructions();
+                Card currentCard = deck.Deal();
+
+                char userInput = GetUserInput(previousCard);
+
+                if (currentCard.CardValue > previousCard.CardValue && userInput == char.ToLower('y') || currentCard.CardValue == previousCard.CardValue && userInput == char.ToLower('n')
+                    || currentCard.CardValue < previousCard.CardValue && userInput == char.ToLower('n'))
+                {
+                    Console.WriteLine($"\n{spacing}Correct! The next card is {currentCard}.");
+
```

Comment:
I really like your messages to the user! Nice Job!

Reply:
Thank you, thank you.